pinot-core/src/main/java/org/apache/pinot/core/util/SchemaUtils.java

```diff
@@ -41,28 +39,6 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(SchemaUtils.class);
 
-  /**
-   * Extracts the source fields and destination fields from the schema
-   * For field specs with a transform expression defined, use the arguments provided to the function
-   * By default, add the field spec name
-   *
-   * TODO: for now, we assume that arguments to transform function are in the source i.e. there's no columns which are derived from transformed columns
-   */
-  public static Set<String> extractSourceFields(Schema schema) {
```

Comment:
Spark/Hadoop jobs that depend on this method may break. Consider deprecating it in stead of removing it. Or, atleast add a backward incompat

Reply:
Added backward incompat label, and added to release notes section